addons/html_builder/static/src/plugins/google_map_option/google_map_option.js

```diff
@@ -0,0 +1,49 @@
+import { useRef, onWillStart, onMounted } from "@odoo/owl";
+import { BaseOptionComponent, useDomState } from "@html_builder/core/utils";
+
+export class GoogleMapOption extends BaseOptionComponent {
+    static template = "html_builder.GoogleMapOption";
+    static props = {
+        loadGoogleMaps: { type: Function },
+        commitPlace: { type: Function },
+    };
+
+    async setup() {
+        super.setup();
+        this.inputRef = useRef("inputRef");
+        this.state = useDomState(map => ({ map }));
+        onWillStart(async () => {
+            await this.props.loadGoogleMaps(this.state.map);
+        });
+        onMounted(() => {
+            this.initializeAutocomplete(this.inputRef.el);
+        });
+    }
+
+    /**
+     * Initialize Google Places API's autocompletion on the option's input.
+     *
+     * @param {Element} inputEl
+     */
+    initializeAutocomplete(inputEl) {
+        if (!this.googleMapsAutocomplete && google?.maps?.places) {
+            this.googleMapsAutocomplete = new google.maps.places.Autocomplete(
+                inputEl,
+                { types: [ 'geocode' ] },
+            );
+            google.maps.event.addListener(
+                this.googleMapsAutocomplete,
+                'place_changed',
+                this.onPlaceChanged.bind(this),
+            );
```

Comment:
I m not sure it will work with the history. I don t know what is done by google.maps.places.Autocomplete

Reply:
It's initializing the API, attaching it to the input element. This is in the sidebar option though, does the history have something to do with it?